vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -129,12 +140,23 @@ public void handle(RoutingContext context) {
         path = indexPage;
       }
 
+      // Access fileSystem once here to be safe
+      FileSystem fs = fileSystem;
+      if (fs == null) {
+        if (classLoader == null) {
+          fs = context.vertx().fileSystem();
```

Comment:
Ah, ok, I see. Here it will use the event loop TCCL right? And you can't control it.

Reply:
Actually, your approach is better. More explicit.